pycbc/distributions/sampling_from_config.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (C) 2021  Shichao Wu
+#
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; If not, see <http://www.gnu.org/licenses/>.
+
+
+import argparse
+import numpy as np
+from pycbc.workflow import WorkflowConfigParser, configuration
+import pycbc.distributions as distributions
+from pycbc.coordinates import spherical_to_cartesian
+from pycbc.cosmology import redshift_from_comoving_volume, distance_from_comoving_volume
+
+
+def draw_samples_from_config(config_path, samples_num=1, seed=150914, **kwds):
+    """ Generate sampling points from a standlone .ini file.
+    Parameters
+    ----------
+    config_path : str
+        The path of the .ini file.
+    samples_num : int
+        The number of samples.
+    seed: int
+        The random seed for sampling.
+    Returns
+    --------
+    samples : list
+        The parameter values [{param:value}] of sampling points in the parameter space.
+    """
+
+    np.random.seed(seed)
+    parser = argparse.ArgumentParser(description=__doc__,
+                formatter_class=argparse.RawDescriptionHelpFormatter)
+    configuration.add_workflow_command_line_group(parser)
+    parser.add_argument('--dist-section', default='prior',
```

Comment:
Why are you setting up a parser and adding options here? How would one even set them? I think you can probably remove this no and simply pass the few needed options as keyword arguments, no? 

so essentially lines 43-58 should probably be removed and just use a directly passed config file name.